_posts/2019-11-01-deploying-debezium-with-kafkaconnector-resource.md

```diff
@@ -0,0 +1,697 @@
+| products            |
+| products_on_hand    |
++---------------------+
+4 rows in set (0.00 sec)
+```
+
+> Don't worry, this isn't the awesome ascii art.
+
+If you want, you can have a poke around this demo database, but when you're done leave this mySQL client running in its own terminal window, so you can come back to it later.
+
+# Create a Kafka cluster
+
+Now we can follow some of the [Strimzi quickstart](https://strimzi.io/quickstarts/minikube/) to create a Kafka cluster running inside `minikube`.
+
+First start minikube:
+
+```shell
+minikube start --memory=4096
+```
+
+When the command finishes you can create a namespace for the resources we're going to create:
+
+```shell
+kubectl create namespace kafka
+```
+
+Then install the cluster operator and associated resources:
+
+```shell
+curl -L https://github.com/strimzi/strimzi-kafka-operator/releases/download/0.15.0/strimzi-cluster-operator-0.15.0.yaml \
+  | sed 's/namespace: .*/namespace: kafka/' \
+  | kubectl apply -f - -n kafka 
+```
+
+And spin up a Kafka cluster, waiting until it's ready:
+
+```shell
+kubectl -n kafka \
+    apply -f https://raw.githubusercontent.com/strimzi/strimzi-kafka-operator/0.15.0/examples/kafka/kafka-persistent-single.yaml \
+  && kubectl wait kafka/my-cluster --for=condition=Ready --timeout=300s -n kafka
+```
+
+What we've got so far looks like this
+
+<pre style="line-height: 1.2 !important;"><code>
+┌────────────────────────────────┐         ┌────────────────────────────────┐
+│ minikube, namespace: kafka     │         │ docker                         │
+│                                │         │                                │
+│ Kafka                          │         │ MySQL                          │
+│  name: my-cluster              │         │                                │
+│                                │         └────────────────────────────────┘
+│ KafkaConnect                   │
+│  name: my-connect              │
+│                                │
+└────────────────────────────────┘
+</code></pre>
+
+> Don't worry, the ASCII art gets a lot better than this!
+
+What's missing from this picture is Kafka Connect in the minikube box.
+
+# Kafka Connect image
+
+The next step is to [create a Strimzi Kafka Connect image](https://strimzi.io/docs/master/#creating-new-image-from-base-str) with the Debezium connector jar file. 
+
+First download and extract the Debezium MySQL connector archive
+```shell
+curl https://repo1.maven.org/maven2/io/debezium/debezium-connector-mysql/0.10.0.Final/debezium-connector-mysql-0.10.0.Final-plugin.tar.gz \
+| tar xvz
+```
+
+Prepare a `Dockerfile` which adds those connector files to the Strimzi Kafka Connect image
+
+```shell
+cat <<EOF >Dockerfile
+FROM strimzi/kafka:latest-kafka-2.3.0
+USER root:root
+EXEC mkdir -p /opt/kafka/plugins/debezium
+COPY ./debezium-connector-mysql/ /opt/kafka/plugins/debezium/
+USER 1001
+EOF
+```
+
+Then build the image from that `Dockerfile` and push it to dockerhub.
```

Comment:
Is this remote push strictly required, or can it be done locally, too? If needed indeed, a sentence is warranted that an account is needed, but it adds quite a bit of friction.

Reply:
It is possible to do this (e.g. [`eval $(minikube docker-env)`](https://stackoverflow.com/questions/42564058/how-to-use-local-docker-images-with-minikube#42564211)), but that method doesn't work because we have no control over the imagePullPolicy. Other ways of achieving the same thing are more complex, so in the end I felt that going via dockerhub, while not ideal, was simple and guaranteed to work. Maybe @scholzj knows of a better way of doing it?